soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -872,12 +882,23 @@ def get_host_ip():
     fw_file = None
 
 if args.server_addr:
-    HOST = args.server_addr
+    url = urlparse("//" + args.server_addr)
+    HOST, _, PORT = url.netloc.partition(':')
```

Comment:
No, you must use `.hostname` and `.port`, you must not use `.netloc`. `urlparse` does all the parsing for you, do not second-guess `urlparse`

```python
# BAD
>>> urllib.parse.urlparse('//[::1]').netloc
'[::1]'
>>> urllib.parse.urlparse('//[::1]').netloc.partition(':')
('[', ':', ':1]') # BAD!
```

```python
>>> urllib.parse.urlparse('//[::1]').hostname
'::1'
>>> urllib.parse.urlparse('//[::1]').port
None
```

Reply:
It is much better, got it fixed, thanks.